### DISCUSSION TITLE
Free trials with Stripe

### DISCUSSION THREAD

**Discussion by garethfr** [pay-rails/pay#563] [Q&A] [answered]
I'm implementing free trials with stripe, and considering the following options:

- creating the free trial with stripe as the payment processor. The issue with this is that when the user subscribes, they have to update their payment info on the free trial subscription, rather than creating a new subscription. This means that the choice of payment package (monthly, yearly etc) would have to be made initially, when the free trial is activated, rather than when the user pays. (OK the user can modify this in stripe, but it's not a smooth process compared to one click from our website)

- Create a free trial locally with the fake processor. The issue with this is that when the user subscribes with stripe, I will need to pass the remaining free trial days to stripe during the checkout, and I will have 2 subscriptions for the user locally. I'm not sure how pay-rails would deal with this situation.

If anyone has implemented anything similar, could they give any advice?

**Comment by excid3** [comment, marked as answer]
I think the second option is smoothest for users.

You would want to cancel the fake processor subscription when they successfully subscribe with Stripe.

**Reply by garethfr** [reply]
thanks for the tip, I'll try it out and report back
